server/src/services/UnsubscribeToken.ts

```diff
@@ -0,0 +1,32 @@
+import { sign } from 'jsonwebtoken';
+
+import { getConfig } from '../config';
+
+export enum unsubscribeType {
+  Chapter,
+  Event,
+}
+
+export type UnsubscribeToken = {
+  type: unsubscribeType;
+  id: number;
+  userId: number;
+  iat: number;
+  exp: number;
+};
+
+export const generateToken = (
+  type: unsubscribeType,
+  id: number,
+  userId: number,
+) => {
+  return sign(
+    {
+      type: type,
+      id: id,
+      userId: userId,
+    },
+    getConfig('JWT_SECRET'),
+    { expiresIn: '14d' },
+  );
+};
```

Comment:
```suggestion
) => sign({ type, id, userId }, getConfig('EMAIL_SECRET'));
```
For this, it's very important that we use a different secret than for the bearer tokens.  Reason being, a user could easily share an unsubscribe token by accident (they email an event invite to a friend, say). If the token is signed with this secret, then that user just shared a bearer token and their friend can use that to login as them.  Kinda dangerous.

Given that I almost didn't notice, I'm going to rename `JWT_SECRET` to something less generic to save my future self.

Aside from that, once we're confident that the token can *only* be used to unsub, then I think it should never expire.  Reason being: we have to (legally, I believe) make it easy to unsubscribe and we don't want our emails reporting as spam.  Most people will click on a recent email, sure, but you never know.

Finally, this is a little neater!

Reply:
This sounds good as well!